dingo/gw/noise/synthetic_noise/generate_dataset.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import copy
+import textwrap
+import numpy as np
+from typing import Dict
+
+import yaml
+
+from dingo.gw.dataset import DingoDataset
+from dingo.gw.noise.asd_dataset import ASDDataset
+from dingo.gw.noise.synthetic_noise.asd_parameterization import parameterize_asd_dataset
+from dingo.gw.noise.synthetic_noise.asd_sampling import KDE, get_rescaling_params
+from dingo.gw.noise.synthetic_noise.utils import reconstruct_psds_from_parameters
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=textwrap.dedent(
+            """\
+        Generate a synthetic noise ASD dataset from an existing dataset of real ASDs.
+        """
+        ),
+    )
+    parser.add_argument(
+        "--asd_dataset",
+        type=str,
+        required=True,
+        help="Path to existing ASD dataset to be parameterized and re-sampled",
+    )
+    parser.add_argument(
+        "--settings_file",
+        type=str,
+        required=True,
+        help="YAML file containing database settings",
+    )
+    parser.add_argument(
+        "--num_processes",
+        type=int,
+        default=1,
+        help="Number of processes to use in pool for parallel parameterization",
+    )
+    parser.add_argument(
+        "--out_file",
+        type=str,
+        default="synthetic_asd_dataset.hdf5",
+        help="Name of file for storing dataset.",
+    )
+    parser.add_argument("--verbose", action="store_true")
+
+    return parser.parse_args()
+
+
+def generate_dataset(real_dataset, settings: Dict, num_processes: int, verbose: bool):
+    parameters_dict = parameterize_asd_dataset(
+        real_dataset,
+        settings["parameterization_settings"],
+        num_processes,
+        verbose=verbose,
+    )
+
+    synthetic_dataset_dict = copy.deepcopy(real_dataset.to_dictionary())
+    synthetic_dataset_dict["settings"]["parameterization_settings"] = settings[
+        "parameterization_settings"
+    ]
+
+    sampling_settings = settings.get("sampling_settings", None)
```

Comment:
The default second argument to `dict.get` is `None`.

```suggestion
    sampling_settings = settings.get("sampling_settings")
```